io/zenoh-transport/src/unicast/universal/link.rs

```diff
@@ -38,58 +38,66 @@ use zenoh_sync::{RecyclingObject, RecyclingObjectPool, Signal};
     // The transport this link is associated to
     transport: TransportUnicastUniversal,
     // The signals to stop TX/RX tasks
-    handle_tx: Option<Arc<async_executor::Task<()>>>,
+    handle_tx: Option<Arc<JoinHandle<()>>>, //Option<Arc<async_executor::Task<()>>>,
     signal_rx: Signal,
     handle_rx: Option<Arc<JoinHandle<()>>>,
 }
 
 impl TransportLinkUnicastUniversal {
-    pub(super) fn new(transport: TransportUnicastUniversal, link: TransportLinkUnicast) -> Self {
-        Self {
-            link,
-            pipeline: None,
-            transport,
-            handle_tx: None,
-            signal_rx: Signal::new(),
-            handle_rx: None,
-        }
+    pub(super) fn new(
+        transport: TransportUnicastUniversal,
+        link: TransportLinkUnicast,
+        priority_tx: &[TransportPriorityTx],
+    ) -> (Self, TransmissionPipelineConsumer) {
+        assert!(!priority_tx.is_empty());
+
+        let config = TransmissionPipelineConf {
+            is_streamed: link.link.is_streamed(),
+            #[cfg(feature = "transport_compression")]
+            is_compression: link.config.is_compression,
+            batch_size: link.config.mtu,
+            queue_size: transport.manager.config.queue_size,
+            backoff: transport.manager.config.queue_backoff,
+        };
+        // The pipeline
+        let (producer, consumer) = TransmissionPipeline::make(config, priority_tx);
+
+        (
+            Self {
+                link: Arc::new(link),
+                pipeline: Some(producer),
+                transport,
+                handle_tx: None,
+                signal_rx: Signal::new(),
+                handle_rx: None,
+            },
+            consumer,
+        )
     }
 }
 
 impl TransportLinkUnicastUniversal {
     pub(super) fn start_tx(
         &mut self,
-        executor: &TransportExecutor,
+        consumer: TransmissionPipelineConsumer,
+        _executor: &TransportExecutor,
         keep_alive: Duration,
-        priority_tx: &[TransportPriorityTx],
     ) {
         if self.handle_tx.is_none() {
-            let config = TransmissionPipelineConf {
-                is_streamed: self.link.link.is_streamed(),
-                #[cfg(feature = "transport_compression")]
-                is_compression: self.link.config.is_compression,
-                batch_size: self.link.config.mtu,
-                queue_size: self.transport.manager.config.queue_size,
-                backoff: self.transport.manager.config.queue_backoff,
-            };
-
-            // The pipeline
-            let (producer, consumer) = TransmissionPipeline::make(config, priority_tx);
-            self.pipeline = Some(producer);
-
             // Spawn the TX task
-            let c_link = self.link.clone();
+            let mut tx = self.link.tx();
             let c_transport = self.transport.clone();
-            let handle = executor.spawn(async move {
+            let handle = async_std::task::spawn(async move {
```

Comment:
You need to spawn the tx thread in the `TransportExecutor`, not in the `async_std` global one.

Reply:
That is a problem that I mentioned - the task spawned in the executor seems not to be executed. Sounds strange, but that's it, I'm working on how to understand why that happens